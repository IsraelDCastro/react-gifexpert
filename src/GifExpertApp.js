import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export default function GifExpertApp () {
  const [categories, setCategories] = useState(['One Piece']);

  // const handleAdd = () => {
    // setCategories('HunterXHunter', ...categories);
    // setCategories(categories => [...categories, 'HunterXHunter']);
  // }
  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={ setCategories } />
      <hr/>
      <ol>
        { categories.map((category) => (
            <GifGrid key={category} category={category} />
          )
        )}
      </ol>
    </>
  )
}
