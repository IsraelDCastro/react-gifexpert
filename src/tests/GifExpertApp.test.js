import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Tests in <GifExpertApp />", () => {
  test("must be rendered correctly", () => {
    const wrapper = shallow(<GifExpertApp/>);

    expect(wrapper).toMatchSnapshot();
  });

  test("must be a list categories", () => {
    const categories = ['One Piece', 'Boku no Hero Academy', 'Full Metal Archemist Brotherhood'];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
