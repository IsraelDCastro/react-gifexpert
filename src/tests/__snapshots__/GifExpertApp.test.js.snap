// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> must be rendered correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
